packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -234,28 +251,60 @@ const MainRoot = React.forwardRef((props, forwardedRef) => {
 // ----------------------------------------------------------------------------
 
 const Root = React.forwardRef((props, forwardedRef) => {
-  const { theme, children, themeOptions, className, ...rest } = props;
+  const {
+    theme,
+    children: childrenProp,
+    themeOptions,
+    className,
+    ...rest
+  } = props;
 
   const prefersDark = useMediaQuery('(prefers-color-scheme: dark)');
   const prefersHighContrast = useMediaQuery('(prefers-contrast: more)');
   const shouldApplyDark = theme === 'dark' || (theme === 'os' && prefersDark);
   const shouldApplyHC = themeOptions?.highContrast ?? prefersHighContrast;
   const shouldApplyBackground = themeOptions?.applyBackground;
 
-  return (
-    <Box
-      className={cx(
-        'iui-root',
-        { 'iui-root-background': shouldApplyBackground },
-        className,
-      )}
-      data-iui-theme={shouldApplyDark ? 'dark' : 'light'}
-      data-iui-contrast={shouldApplyHC ? 'high' : 'default'}
-      ref={forwardedRef}
-      {...rest}
+  const children = React.useMemo(
+    () => (
+      <Box
+        className={cx(
+          'iui-root',
+          { 'iui-root-background': shouldApplyBackground },
+          className,
+        )}
+        data-iui-theme={shouldApplyDark ? 'dark' : 'light'}
+        data-iui-contrast={shouldApplyHC ? 'high' : 'default'}
+        data-iui-bridge={
+          themeOptions?.bridgeToFutureVersions ? true : undefined
+        }
+        ref={forwardedRef}
+        {...rest}
+      >
+        {childrenProp}
+      </Box>
+    ),
+    [
+      childrenProp,
+      className,
+      forwardedRef,
+      rest,
+      shouldApplyBackground,
+      shouldApplyDark,
+      shouldApplyHC,
+      themeOptions?.bridgeToFutureVersions,
+    ],
+  );
+
+  return themeOptions?.bridgeToFutureVersions ? (
+    <ITwinUIV5Root
+      colorScheme={shouldApplyDark ? 'dark' : 'light'}
+      density='dense'
```

Comment:
Continuing the same comment, this `Root` should be set up by consumers to wrap around iTwinUI v3's `ThemeProvider`. We should have well-written instructions on how to set it up.

Reply:
> `react-workshop` now depends on `iTwinUI-react@5.0.0-alpha.0` and sets up `<ITwinUIV5Root>`/`<Root>` on its end. ([a216d1b](https://github.com/iTwin/iTwinUI/commit/a216d1b0f3e2a1386a7a2444c6987f19626b949b)).

`<Root>` should be wrapped around `<ThemeProvider>`, not the other way around, otherwise the v3 variables will take preference.
 
> Regarding the "well-written instructions", I'm not sure where we want to document everything related to this theme bridge. Once we decide a place, I will add those instructions. For now, added a checklist item in the PR description.

We can start with a page in the github wiki, and move it elsewhere later if necessary.

Reply:
> shall I create a WIP GitHub wiki for it?

Yeah, let's do it now.